tests/time_tests/src/timetests_helper/statistics_writer.h

```diff
@@ -46,12 +49,38 @@ class StatisticsWriter {
     }
   }
 
+  /**
+   * @brief Compute order for statistics operations.
+   */
+  void addOrderCount() {
+    order_count++;
+  }
+
+  void deleteOrderCount() {
+    order_count--;
+  }
+
+  /**
+   * @brief Writes statistics in map structure.
+   */
+  void addToTimeStructure(const std::pair<std::string, float> &record, int &order_number) {
```

Comment:
`addToTimeStructure` always called together with `deleteOrderCount`, so I see no reason to have `order_number` at all. Let's collect number of `Timer` creations by one counter and timers nesting by other counter